internal/repository/postgres/tenant.go

```diff
@@ -17,26 +19,43 @@ func NewTenantRepository(db *postgres.DB, logger *logger.Logger) tenant.Reposito
 	return &tenantRepository{db: db, logger: logger}
 }
 
-func (r *tenantRepository) Create(ctx context.Context, tenant *tenant.Tenant) error {
+func (r *tenantRepository) Create(ctx context.Context, t *tenant.Tenant) error {
 	query := `
+        SELECT 1 FROM tenants where name = $1
+    `
+	var exists bool
+	err := r.db.QueryRowContext(
+		ctx,
+		query,
+		t.Name,
+	).Scan(&exists)
+	if err != nil && err != sql.ErrNoRows {
+		return err
+	}
+
+	if exists {
+		return ierr.New(ierr.ErrAlreadyExists, "tenant already exists")
+	}
```

Comment:
Although the change is fine - would recommend to not pick these improvements on the way
We already have a PR open for moving tenant, user and env domains to ent so it handles all these things out fo the box

Reply:
got it